### PR TITLE
[WIP] Enable IPv6 first cluster test

### PR DESCRIPTION
**Pull request by zirain**
WIP, use CI to test some pending PRs together.

### CODE REVIEW

**Comment by zhaohuabing** [comment]
@zirain should this be in v1.2.0 ?

**Comment by zirain** [comment]
> @zirain should this be in v1.2.0 ?

therer're a lot of pending PRs that block this one.

File: internal/infrastructure/kubernetes/proxy/resource.go [hunk 1]
```diff
@@ -135,6 +136,12 @@ func expectedProxyContainers(infra *ir.ProxyInfra,
 		},
 		MaxHeapSizeBytes: maxHeapSizeBytes,
 		XdsServerHost:    ptr.To(fmt.Sprintf("%s.%s.svc.%s", config.EnvoyGatewayServiceName, namespace, dnsDomain)),
+		IPFamily:         net.PreferIPFamily(ipv6First, infra.Config),
+	}
+
+	// TODO: this's a temporary solution to support IPv6 first cluster for Envoy Gateway.
```
Comment (zirain) [inline internal/infrastructure/kubernetes/proxy/resource.go:142]: Remind myself not to forget to send a separate PR with modification here.
